alfresco-search/src/main/java/org/alfresco/solr/AlfrescoSolrDataModel.java

```diff
@@ -800,8 +800,9 @@ else if ((propertyDefinition.getIndexTokenisationMode() == IndexTokenisationMode
    
     private void addFullTextSearchFields( PropertyDefinition propertyDefinition , IndexedField indexedField)
     {
-        if ((propertyDefinition.getIndexTokenisationMode() == IndexTokenisationMode.TRUE)
+        if (((propertyDefinition.getIndexTokenisationMode() == IndexTokenisationMode.TRUE)
                 || (propertyDefinition.getIndexTokenisationMode() == IndexTokenisationMode.BOTH))
+                && !identifierProperties.contains(propertyDefinition.getName()))

```

Comment:
Do I understand this correctly that the value for the tokenised attribute of a property will be ignored if the property is set as an identifier? If that's the case I'm not so sure if this is the right way to fix the issue.